include/ringmesh/mesh/cartesian_grid.h

```diff
@@ -281,11 +281,160 @@ namespace RINGMesh
          * @return the attribute manager containing the grid point attributes
          * values.
          */
-        GEO::AttributesManager& attributes_manager() const
+        GEO::AttributesManager& attributes_manager()
         {
             return attributes_manager_;
         }
 
+        /*!
+         * @return the attribute manager containing the grid point attributes
+         * values.
+         */
+        const GEO::AttributesManager& attributes_manager() const
+        {
+            return attributes_manager_;
+        }
+
+        void add_index_t_attribute( const std::string& attribute_name,
+            const std::vector< index_t >& values )
+        {
+            if( values.size() != nb_total_cells_ )
+            {
+                throw RINGMeshException( "CartesianGrid",
+                    "The attribute you're trying to add doesn't have the same"
+                    " number of values than the grid." );
+            }
+            attributes_manager_.bind_attribute_store( attribute_name,
+                GEO::AttributeStore::
+                    create_attribute_store_by_element_type_name(
+                        "index_t", 1 ) );
+            GEO::Attribute< index_t > attribute{ attributes_manager_,
+                attribute_name };
+            for( auto i : range( nb_total_cells_ ) )
+            {
+                attribute[i] = values[i];
+            }
+        }
+
+        void add_float_attribute( const std::string& attribute_name,
+            const std::vector< float >& values )
+        {
+            if( values.size() != nb_total_cells_ )
+            {
+                throw RINGMeshException( "CartesianGrid",
+                    "The attribute you're trying to add doesn't have the same"
+                    " number of values than the grid." );
+            }
+            attributes_manager_.bind_attribute_store( attribute_name,
+                GEO::AttributeStore::
+                    create_attribute_store_by_element_type_name( "float", 1 ) );
+            GEO::Attribute< float > attribute{ attributes_manager_,
+                attribute_name };
+            for( auto i : range( nb_total_cells_ ) )
+            {
```

Comment:
there is a funtion fill in the geo::attrinute why dont using it?

Reply:
Because this function gives a different value for each point of the grid, and fill gives the same attribute value to all the points (I do use it when it is needed, 14 lines under that)